In [2]:
import json
from urllib.request import urlopen

def get_top200(country):

    url = 'https://rss.itunes.apple.com/api/v1/' + country + '/podcasts/top-podcasts/all/200/explicit.json'
    with urlopen(url) as data_file:    
        data = json.load(data_file)

    results = data["feed"]    
    return results

In [3]:
import csv

def writePodcastsToFile(country, results):
    fields = ['country', 'rank','name', 'id', 'releaseDate', 'Primary_Genre', 'Other_Genre' , 'Artist']

    filename = "top_GB_podcasts.csv"

    allofthem = []
    cntr = 0

    for item in results["results"]:
        gnr = item["genres"][0]
        all_genres = ""
        all_genres_length = len(item["genres"])
        for x in range(1,all_genres_length):
            other_gnr =  item["genres"][x]
            other_name = other_gnr["name"]
            if other_name != 'Podcasts':
                all_genres +=  other_name + ","

        cntr += 1
        x = [country,cntr, item["name"], item["id"],  item["releaseDate"],gnr["name"], all_genres, item["artistName"]]
        allofthem.append(x)


    with open(filename, 'a', newline ='',encoding="utf-8") as csvfile: 
        writer = csv.writer(csvfile) 
        writer.writerow(fields) 
        writer.writerows(allofthem) 

In [4]:
countries =  [  'gb' ] #['us','za','uk', 'au','in','ir',] 

for c in countries:
    r = get_top200(c)
    writePodcastsToFile(c, r)

In [164]:
import podsearch
#podcasts = podsearch.search("Deciding Decade", country="us", limit=1) 1525423535 
podcasts = podsearch.search("1135137367", country="us", limit=1)

#https://itunes.apple.com/lookup?id=1135137367&country=US&media=podcast&entity=podcastEpisode&limit=1
    
print ("NPR items are :::::::::::::::::::::::")

for item in podcasts:
    print (item.name, item.id, item.author, item.episode_count)
    

NPR items are :::::::::::::::::::::::
The Tim Dillon Show 1135137367 Tim Dillon 97


In [29]:
import csv

all_podcasts = []
with open("top_podcasts.csv", 'r',encoding="utf-8") as csvfile:
    csvreader = csv.reader(csvfile)
    fields = next(csvreader)
    
    for row in csvreader:
        all_podcasts.append([row[0],row[3]])

1059


In [35]:
temp = all_podcasts[5:12]


In [37]:
#import podsearch
import urllib.request
from urllib.request import HTTPError

def getEpisodeCounts(PodcastList):
    episode_count = []
    fields = [ 'name', 'id', 'Artist', 'episode_count']
    filename = "episode_counts.csv"

    url_part1 = "https://itunes.apple.com/lookup?id="
    url_part2 = "&country="
    url_part3 = "&media=podcast&entity=podcastEpisode&limit=1/json"

    pod_id = 0
    for eachitem in PodcastList:
        pod_id = str(eachitem[1])
        cntry =  str(eachitem[0])

        try:
            url = url_part1 + pod_id + url_part2 + cntry + url_part3
            #print (url)
            with urlopen(url) as data_file:    
                reviewdata = json.load(data_file)
                x = reviewdata['results'] 
                xx = x[0]
                tc = xx['trackCount'] 
                athr = xx['artistName']
                name = xx['collectionName']
           # for sub_item in podcasts:
           #     x = [sub_item.name, sub_item.id, sub_item.author, sub_item.episode_count]
                episode_count.append([name, pod_id ,athr,tc])
        except Exception as err:
            print (cntry, pod_id, err)
            pass

    with open(filename, 'a', newline ='') as csvfile: 
        writer = csv.writer(csvfile) 
        writer.writerow(fields) 
        writer.writerows(episode_count)
    ################################  

In [38]:
getEpisodeCounts(all_podcasts)

us 485351658 list index out of range


In [47]:
print (len(all_podcasts))

1059


# This is where the real magic starts.  The all_podcasts variable is already populated with 1059 podcast_ID and country_code.  Run from the following cells

In [42]:
all_podcasts = [['us', '1540849480'], ['us', '1322200189'], ['us', '1200361736'], ['us', '1464919521'], ['us', '1047335260'], ['us', '1539568321'], ['us', '1549903604'], ['us', '1222114325'], ['us', '1552069504'], ['us', '1553335461'], ['us', '1249486443'], ['us', '278981407'], ['us', '77001367'], ['us', '1553089886'], ['us', '1480311435'], ['us', '1550331348'], ['us', '1554132477'], ['us', '1548604447'], ['us', '1532894981'], ['us', '934820588'], ['us', '965293227'], ['us', '917918570'], ['us', '201671138'], ['us', '1091709555'], ['us', '1494350511'], ['us', '1028908750'], ['us', '1552058389'], ['us', '214089682'], ['us', '1551162705'], ['us', '1541330058'], ['us', '1503226625'], ['us', '1450994021'], ['us', '1500667648'], ['us', '1542475843'], ['us', '290783428'], ['us', '1205030005'], ['us', '294055449'], ['us', '1543456877'], ['us', '1484511465'], ['us', '1057255460'], ['us', '1537836233'], ['us', '1122804248'], ['us', '1528594034'], ['us', '1547645093'], ['us', '929121341'], ['us', '1334878780'], ['us', '1468013270'], ['us', '1232428553'], ['us', '1460600818'], ['us', '863897795'], ['us', '1549814799'], ['us', '306390087'], ['us', '1479234915'], ['us', '354668519'], ['us', '1150510297'], ['us', '1232009882'], ['us', '275699983'], ['us', '594419649'], ['us', '1184022695'], ['us', '1087110764'], ['us', '1553793022'], ['us', '1266577963'], ['us', '1119389968'], ['us', '1524250628'], ['us', '620967489'], ['us', '1487513861'], ['us', '1367210511'], ['us', '1532956108'], ['us', '209377688'], ['us', '1545953110'], ['us', '1454587745'], ['us', '523121474'], ['us', '160904630'], ['us', '1553782780'], ['us', '1450211392'], ['us', '1482310665'], ['us', '1435516849'], ['us', '1440833267'], ['us', '1435183458'], ['us', '1264843400'], ['us', '1551820452'], ['us', '152249110'], ['us', '121493675'], ['us', '1229434818'], ['us', '1087147821'], ['us', '1555769971'], ['us', '283605519'], ['us', '1317493077'], ['us', '1346314086'], ['us', '1434243584'], ['us', '790487079'], ['us', '1305546061'], ['us', '1233834541'], ['us', '1482575855'], ['us', '1033048640'], ['us', '1549704404'], ['us', '1382804627'], ['us', '1441595858'], ['us', '1500477470'], ['us', '1451109634'], ['us', '987967575'], ['us', '173001861'], ['us', '1286818575'], ['us', '1469394914'], ['us', '394775318'], ['us', '1362265400'], ['us', '1524080195'], ['us', '1320118593'], ['us', '1485351658'], ['us', '135067274'], ['us', '1073226719'], ['us', '978052928'], ['us', '1554108046'], ['us', '809264944'], ['us', '137254859'], ['us', '121493804'], ['us', '1433974017'], ['us', '703889772'], ['us', '1042368254'], ['us', '1466667116'], ['us', '1460147025'], ['us', '596047499'], ['us', '1210741497'], ['us', '1537865727'], ['us', '1461511691'], ['us', '732915228'], ['us', '1551644089'], ['us', '1474245040'], ['us', '201853034'], ['us', '1378040733'], ['us', '1495601614'], ['us', '1012570406'], ['us', '1081124929'], ['us', '147247199'], ['us', '1140596919'], ['us', '1510365500'], ['us', '1553377190'], ['us', '1193068130'], ['us', '1112190608'], ['us', '1337195586'], ['us', '1148175292'], ['us', '306106212'], ['us', '1334767397'], ['us', '1490924360'], ['za', '1450994021'], ['za', '160904630'], ['za', '1525254543'], ['za', '1264843400'], ['za', '1334878780'], ['za', '1072307993'], ['za', '1540543705'], ['za', '216015753'], ['za', '137254859'], ['za', '135067274'], ['za', '1486896400'], ['za', '1556449891'], ['za', '1529419354'], ['za', '1033048640'], ['za', '1469703618'], ['za', '1184022695'], ['za', '278981407'], ['za', '1334767397'], ['za', '1068207118'], ['za', '863897795'], ['za', '1322200189'], ['za', '1335814741'], ['za', '921359216'], ['za', '1440251673'], ['za', '1480311435'], ['za', '928933489'], ['za', '1512102345'], ['za', '1494350511'], ['za', '1316294579'], ['za', '1200361736'], ['za', '1484511465'], ['za', '1208024744'], ['za', '1540849480'], ['za', '1552069504'], ['za', '1511485972'], ['za', '1466064652'], ['za', '1522318033'], ['za', '152564333'], ['za', '151230264'], ['za', '1552058389'], ['za', '1487513861'], ['za', '1553335461'], ['za', '1532956108'], ['za', '470622782'], ['za', '1205030005'], ['za', '205578441'], ['za', '1150510297'], ['za', '840986946'], ['za', '833641766'], ['za', '1348831124'], ['za', '1501735272'], ['za', '1397427305'], ['za', '1504738445'], ['za', '1508391516'], ['za', '596047499'], ['za', '1484003537'], ['za', '347854569'], ['za', '1551459171'], ['za', '1547235591'], ['za', '971122157'], ['za', '1433974017'], ['za', '1470375638'], ['za', '1346314086'], ['za', '1028908750'], ['za', '1050832450'], ['za', '1515385282'], ['za', '201671138'], ['za', '1098413063'], ['za', '354668519'], ['za', '1539568321'], ['za', '1191775648'], ['za', '1229434818'], ['za', '1531221910'], ['za', '1554785643'], ['za', '1521841754'], ['za', '1141941405'], ['za', '582272991'], ['za', '364682371'], ['za', '1464919521'], ['za', '1286818575'], ['za', '657476401'], ['za', '1449631195'], ['za', '1552027709'], ['za', '1516194847'], ['za', '917918570'], ['za', '1401057414'], ['za', '538401206'], ['za', '77001367'], ['za', '1505786417'], ['za', '1517954480'], ['za', '1441394221'], ['za', '470846331'], ['za', '1317493077'], ['za', '1119389968'], ['za', '1468107412'], ['za', '1493479119'], ['za', '1495053060'], ['za', '1047335260'], ['za', '1453389736'], ['za', '1122804248'], ['za', '962733842'], ['za', '1550777106'], ['za', '733163012'], ['za', '1322666339'], ['za', '892107837'], ['za', '1545953110'], ['za', '1521469332'], ['za', '1434243584'], ['za', '1416206538'], ['za', '1530452735'], ['za', '218453800'], ['za', '1512968943'], ['za', '451295014'], ['za', '1440133626'], ['za', '1499860465'], ['za', '1039991663'], ['za', '1065559535'], ['za', '1555203383'], ['za', '1333552422'], ['za', '1070649025'], ['za', '1223803641'], ['za', '1475466071'], ['za', '1296350485'], ['za', '1128343449'], ['za', '1550666283'], ['za', '1039481487'], ['za', '1292363943'], ['za', '942777522'], ['za', '1317823455'], ['za', '1453911226'], ['za', '757953988'], ['za', '1547504297'], ['za', '1503921457'], ['za', '1482332890'], ['za', '1458654443'], ['za', '1501213645'], ['za', '1545444200'], ['za', '1352546554'], ['za', '1075744322'], ['za', '73802620'], ['za', '1541330058'], ['za', '1227971746'], ['za', '1353058891'], ['za', '1539535133'], ['za', '394775318'], ['za', '1545841764'], ['za', '1073226719'], ['za', '1507423923'], ['za', '1440833267'], ['za', '1051557000'], ['za', '1462681901'], ['za', '1457336351'], ['za', '1466824259'], ['za', '342735925'], ['za', '1508985962'], ['za', '631094342'], ['za', '1521068608'], ['za', '470623173'], ['za', '1541777977'], ['za', '1512546904'], ['za', '844102142'], ['za', '352660924'], ['za', '334204277'], ['za', '1550649360'], ['za', '1439655378'], ['za', '150892556'], ['za', '978052928'], ['uk', '1540849480'], ['uk', '1322200189'], ['uk', '1200361736'], ['uk', '1464919521'], ['uk', '1047335260'], ['uk', '1539568321'], ['uk', '1549903604'], ['uk', '1222114325'], ['uk', '1552069504'], ['uk', '1553335461'], ['uk', '1249486443'], ['uk', '278981407'], ['uk', '77001367'], ['uk', '1553089886'], ['uk', '1480311435'], ['uk', '1550331348'], ['uk', '1554132477'], ['uk', '1548604447'], ['uk', '1532894981'], ['uk', '934820588'], ['uk', '965293227'], ['uk', '917918570'], ['uk', '201671138'], ['uk', '1091709555'], ['uk', '1494350511'], ['uk', '1028908750'], ['uk', '1552058389'], ['uk', '214089682'], ['uk', '1551162705'], ['uk', '1541330058'], ['uk', '1503226625'], ['uk', '1450994021'], ['uk', '1500667648'], ['uk', '1542475843'], ['uk', '290783428'], ['uk', '1205030005'], ['uk', '294055449'], ['uk', '1543456877'], ['uk', '1484511465'], ['uk', '1057255460'], ['uk', '1537836233'], ['uk', '1122804248'], ['uk', '1528594034'], ['uk', '1547645093'], ['uk', '929121341'], ['uk', '1334878780'], ['uk', '1468013270'], ['uk', '1232428553'], ['uk', '1460600818'], ['uk', '863897795'], ['uk', '1549814799'], ['uk', '306390087'], ['uk', '1479234915'], ['uk', '354668519'], ['uk', '1150510297'], ['uk', '1232009882'], ['uk', '275699983'], ['uk', '594419649'], ['uk', '1184022695'], ['uk', '1087110764'], ['uk', '1553793022'], ['uk', '1266577963'], ['uk', '1119389968'], ['uk', '1524250628'], ['uk', '620967489'], ['uk', '1487513861'], ['uk', '1367210511'], ['uk', '1532956108'], ['uk', '209377688'], ['uk', '1545953110'], ['uk', '1454587745'], ['uk', '523121474'], ['uk', '160904630'], ['uk', '1553782780'], ['uk', '1450211392'], ['uk', '1482310665'], ['uk', '1435516849'], ['uk', '1440833267'], ['uk', '1435183458'], ['uk', '1264843400'], ['uk', '1551820452'], ['uk', '152249110'], ['uk', '121493675'], ['uk', '1229434818'], ['uk', '1087147821'], ['uk', '1555769971'], ['uk', '283605519'], ['uk', '1317493077'], ['uk', '1346314086'], ['uk', '1434243584'], ['uk', '790487079'], ['uk', '1305546061'], ['uk', '1233834541'], ['uk', '1482575855'], ['uk', '1033048640'], ['uk', '1549704404'], ['uk', '1382804627'], ['uk', '1441595858'], ['uk', '1500477470'], ['uk', '1451109634'], ['uk', '987967575'], ['uk', '173001861'], ['uk', '1286818575'], ['uk', '1469394914'], ['uk', '394775318'], ['uk', '1362265400'], ['uk', '1524080195'], ['uk', '1320118593'], ['uk', '1485351658'], ['uk', '135067274'], ['uk', '1073226719'], ['uk', '978052928'], ['uk', '1554108046'], ['uk', '809264944'], ['uk', '137254859'], ['uk', '121493804'], ['uk', '1433974017'], ['uk', '703889772'], ['uk', '1042368254'], ['uk', '1466667116'], ['uk', '1460147025'], ['uk', '596047499'], ['uk', '1210741497'], ['uk', '1537865727'], ['uk', '1461511691'], ['uk', '732915228'], ['uk', '1551644089'], ['uk', '1474245040'], ['uk', '201853034'], ['uk', '1378040733'], ['uk', '1495601614'], ['uk', '1012570406'], ['uk', '1081124929'], ['uk', '147247199'], ['uk', '1140596919'], ['uk', '1510365500'], ['uk', '1553377190'], ['uk', '1193068130'], ['uk', '1112190608'], ['uk', '1337195586'], ['uk', '1148175292'], ['uk', '306106212'], ['uk', '1334767397'], ['uk', '1490924360'], ['au', '94688506'], ['au', '213334232'], ['au', '1553335461'], ['au', '928350893'], ['au', '1217681421'], ['au', '1451469361'], ['au', '1540849480'], ['au', '1503645220'], ['au', '1501500390'], ['au', '1468033698'], ['au', '1461999702'], ['au', '1200361736'], ['au', '1552975779'], ['au', '1554973779'], ['au', '1322200189'], ['au', '278981407'], ['au', '1553858670'], ['au', '1508985962'], ['au', '1482061243'], ['au', '1469153910'], ['au', '1484511465'], ['au', '1476501557'], ['au', '1550543719'], ['au', '1472126503'], ['au', '1549225698'], ['au', '995159486'], ['au', '1146329262'], ['au', '413046628'], ['au', '1555243483'], ['au', '1502954097'], ['au', '1436450976'], ['au', '863897795'], ['au', '1494350511'], ['au', '1102298716'], ['au', '1361401612'], ['au', '1450994021'], ['au', '1533612473'], ['au', '1349523628'], ['au', '1242155321'], ['au', '1212097275'], ['au', '73331325'], ['au', '1552058389'], ['au', '1474052266'], ['au', '1507116817'], ['au', '1033048640'], ['au', '936653355'], ['au', '1117246594'], ['au', '135067274'], ['au', '1385995581'], ['au', '840986946'], ['au', '1545953110'], ['au', '1533629268'], ['au', '160904630'], ['au', '917918570'], ['au', '1552309889'], ['au', '1550688418'], ['au', '201671138'], ['au', '1534252348'], ['au', '73330306'], ['au', '1184022695'], ['au', '1264843400'], ['au', '1205030005'], ['au', '1464919521'], ['au', '73330911'], ['au', '1451864362'], ['au', '1417124389'], ['au', '1451470931'], ['au', '1473523403'], ['au', '1332566161'], ['au', '1435516849'], ['au', '1500477470'], ['au', '1474582435'], ['au', '1150510297'], ['au', '1478211900'], ['au', '1552027709'], ['au', '1333552422'], ['au', '1229434818'], ['au', '1434243584'], ['au', '1553862476'], ['au', '73330961'], ['au', '78995043'], ['au', '1484930949'], ['au', '1260800644'], ['au', '1479973402'], ['au', '969904333'], ['au', '1554782191'], ['au', '360804281'], ['au', '1539790170'], ['au', '1335814741'], ['au', '1210989839'], ['au', '733163012'], ['au', '394775318'], ['au', '1384819415'], ['au', '1524599320'], ['au', '1427018681'], ['au', '1546324192'], ['au', '1500848514'], ['au', '1082244480'], ['au', '1198117246'], ['au', '1487513861'], ['au', '329902285'], ['au', '1028908750'], ['au', '1464488062'], ['au', '1385379989'], ['au', '1548946506'], ['au', '981373142'], ['au', '1296350485'], ['au', '1484760064'], ['au', '637433286'], ['au', '1109943717'], ['au', '1118867383'], ['au', '1534672561'], ['au', '1047335260'], ['au', '344869087'], ['au', '1080505022'], ['au', '1201031401'], ['au', '354668519'], ['au', '1249219269'], ['au', '164330831'], ['au', '1346314086'], ['au', '1480311435'], ['au', '1554132477'], ['au', '73330895'], ['au', '1449500734'], ['au', '842419860'], ['au', '73330343'], ['au', '1454193127'], ['au', '1457667995'], ['au', '256580326'], ['au', '596047499'], ['au', '582272991'], ['au', '1508927190'], ['au', '790487079'], ['au', '173001861'], ['au', '809264944'], ['au', '1541330058'], ['au', '1551120994'], ['au', '1513489117'], ['au', '1481398762'], ['au', '1445344726'], ['au', '1076207687'], ['au', '1507423923'], ['in', '347854569'], ['in', '584092091'], ['in', '160904630'], ['in', '1009197596'], ['in', '1450994021'], ['in', '1494989268'], ['in', '1548244887'], ['in', '1532901566'], ['in', '1036379102'], ['in', '1490896085'], ['in', '1554876547'], ['in', '1548187636'], ['in', '1543887425'], ['in', '1548383228'], ['in', '1119439036'], ['in', '1035148051'], ['in', '888135469'], ['in', '1041684187'], ['in', '1449631195'], ['in', '1445665622'], ['in', '1039644241'], ['in', '1039511252'], ['in', '1552164995'], ['in', '1536206996'], ['in', '1549390988'], ['in', '1448599498'], ['in', '1549822234'], ['in', '1317823455'], ['in', '1521068608'], ['in', '928933489'], ['in', '1550874620'], ['in', '1340196987'], ['in', '1454097755'], ['in', '1437983503'], ['in', '262026989'], ['in', '1065559535'], ['in', '1406730235'], ['in', '1534542817'], ['in', '73802620'], ['in', '928159684'], ['in', '1494440867'], ['in', '1542664721'], ['in', '1334878780'], ['in', '1552604499'], ['in', '1440116479'], ['in', '135067274'], ['in', '1206967377'], ['in', '1209849464'], ['in', '1118875240'], ['in', '979118845'], ['in', '152022135'], ['in', '524002915'], ['in', '278981407'], ['in', '1474429475'], ['in', '1208024744'], ['in', '1522708820'], ['in', '863897795'], ['in', '1523324463'], ['in', '1080974028'], ['in', '151230264'], ['in', '1517640300'], ['in', '659155419'], ['in', '1033048640'], ['in', '1190896090'], ['in', '1131144827'], ['in', '160835381'], ['in', '1200361736'], ['in', '1547246790'], ['in', '1516770387'], ['in', '1536206642'], ['in', '470623037'], ['in', '1456761991'], ['in', '1317493077'], ['in', '1442530475'], ['in', '1184274374'], ['in', '1150510297'], ['in', '1545444200'], ['in', '1227971746'], ['in', '1122804248'], ['in', '1289643592'], ['in', '919631690'], ['in', '1509932636'], ['in', '1512934794'], ['in', '1346314086'], ['in', '1394251788'], ['in', '1437907791'], ['in', '1479332846'], ['in', '1335814741'], ['in', '1487513861'], ['in', '596047499'], ['in', '1138869817'], ['in', '1488735954'], ['in', '1509930659'], ['in', '1509930788'], ['in', '323145903'], ['in', '1068207118'], ['in', '1003811629'], ['in', '1554785643'], ['in', '1481769883'], ['in', '1436719991'], ['in', '1488292026'], ['in', '1438449989'], ['in', '109522474'], ['in', '1377845840'], ['in', '842818711'], ['in', '1526452361'], ['in', '1535001986'], ['in', '1469394914'], ['in', '1474245040'], ['in', '1184022695'], ['in', '1532263528'], ['in', '1494771324'], ['in', '1538630420'], ['in', '1453280018'], ['in', '1391332856'], ['in', '285260960'], ['in', '354668519'], ['in', '833641766'], ['in', '1239293215'], ['in', '1504664694'], ['in', '394775318'], ['in', '1515378138'], ['in', '1508721102'], ['in', '1528481399'], ['in', '1472106563'], ['in', '1545953110'], ['in', '1550568878'], ['in', '1031590716'], ['in', '1213337541'], ['in', '1515173768'], ['in', '262026947'], ['in', '733163012'], ['in', '420929545'], ['in', '263170419'], ['in', '1545333588'], ['in', '1547086264'], ['in', '1466824259'], ['in', '990149481'], ['in', '1449500734'], ['in', '1456538451'], ['in', '1465487889'], ['in', '1542452346'], ['in', '540187826'], ['in', '1216260422'], ['in', '1346062517'], ['in', '1430315931'], ['in', '1488201401'], ['in', '1121239439'], ['in', '1492486688'], ['in', '1247628265'], ['in', '1028908750'], ['in', '1544098624'], ['in', '1480311435'], ['in', '1333552422'], ['in', '279428154'], ['in', '1484511465'], ['in', '1527740349'], ['in', '470624027'], ['in', '1539568321'], ['in', '1502261788'], ['in', '1296350485'], ['in', '969076668'], ['in', '1502692172'], ['in', '137254859'], ['in', '418685623'], ['in', '558494752'], ['in', '1434243584'], ['in', '1322200189'], ['in', '290783428'], ['in', '1495067186'], ['in', '470622782'], ['in', '1553199762'], ['in', '549899114'], ['in', '1019576853'], ['in', '1494350511'], ['in', '325404506'], ['in', '1532956108'], ['in', '205892240'], ['in', '582585399'], ['in', '1201248091'], ['in', '1137521932'], ['in', '1534173216'], ['in', '1547922859'], ['in', '313848583'], ['in', '1472370879'], ['in', '1209219220'], ['in', '1245120783'], ['in', '1531221910'], ['ir', '1540849480'], ['ir', '1322200189'], ['ir', '1200361736'], ['ir', '1464919521'], ['ir', '1047335260'], ['ir', '1539568321'], ['ir', '1549903604'], ['ir', '1222114325'], ['ir', '1552069504'], ['ir', '1553335461'], ['ir', '1249486443'], ['ir', '278981407'], ['ir', '77001367'], ['ir', '1553089886'], ['ir', '1480311435'], ['ir', '1550331348'], ['ir', '1554132477'], ['ir', '1548604447'], ['ir', '1532894981'], ['ir', '934820588'], ['ir', '965293227'], ['ir', '917918570'], ['ir', '201671138'], ['ir', '1091709555'], ['ir', '1494350511'], ['ir', '1028908750'], ['ir', '1552058389'], ['ir', '214089682'], ['ir', '1551162705'], ['ir', '1541330058'], ['ir', '1503226625'], ['ir', '1450994021'], ['ir', '1500667648'], ['ir', '1542475843'], ['ir', '290783428'], ['ir', '1205030005'], ['ir', '294055449'], ['ir', '1543456877'], ['ir', '1484511465'], ['ir', '1057255460'], ['ir', '1537836233'], ['ir', '1122804248'], ['ir', '1528594034'], ['ir', '1547645093'], ['ir', '929121341'], ['ir', '1334878780'], ['ir', '1468013270'], ['ir', '1232428553'], ['ir', '1460600818'], ['ir', '863897795'], ['ir', '1549814799'], ['ir', '306390087'], ['ir', '1479234915'], ['ir', '354668519'], ['ir', '1150510297'], ['ir', '1232009882'], ['ir', '275699983'], ['ir', '594419649'], ['ir', '1184022695'], ['ir', '1087110764'], ['ir', '1553793022'], ['ir', '1266577963'], ['ir', '1119389968'], ['ir', '1524250628'], ['ir', '620967489'], ['ir', '1487513861'], ['ir', '1367210511'], ['ir', '1532956108'], ['ir', '209377688'], ['ir', '1545953110'], ['ir', '1454587745'], ['ir', '523121474'], ['ir', '160904630'], ['ir', '1553782780'], ['ir', '1450211392'], ['ir', '1482310665'], ['ir', '1435516849'], ['ir', '1440833267'], ['ir', '1435183458'], ['ir', '1264843400'], ['ir', '1551820452'], ['ir', '152249110'], ['ir', '121493675'], ['ir', '1229434818'], ['ir', '1087147821'], ['ir', '1555769971'], ['ir', '283605519'], ['ir', '1317493077'], ['ir', '1346314086'], ['ir', '1434243584'], ['ir', '790487079'], ['ir', '1305546061'], ['ir', '1233834541'], ['ir', '1482575855'], ['ir', '1033048640'], ['ir', '1549704404'], ['ir', '1382804627'], ['ir', '1441595858'], ['ir', '1500477470'], ['ir', '1451109634'], ['ir', '987967575'], ['ir', '173001861'], ['ir', '1286818575'], ['ir', '1469394914'], ['ir', '394775318'], ['ir', '1362265400'], ['ir', '1524080195'], ['ir', '1320118593'], ['ir', '1485351658'], ['ir', '135067274'], ['ir', '1073226719'], ['ir', '978052928'], ['ir', '1554108046'], ['ir', '809264944'], ['ir', '137254859'], ['ir', '121493804'], ['ir', '1433974017'], ['ir', '703889772'], ['ir', '1042368254'], ['ir', '1466667116'], ['ir', '1460147025'], ['ir', '596047499'], ['ir', '1210741497'], ['ir', '1537865727'], ['ir', '1461511691'], ['ir', '732915228'], ['ir', '1551644089'], ['ir', '1474245040'], ['ir', '201853034'], ['ir', '1378040733'], ['ir', '1495601614'], ['ir', '1012570406'], ['ir', '1081124929'], ['ir', '147247199'], ['ir', '1140596919'], ['ir', '1510365500'], ['ir', '1553377190'], ['ir', '1193068130'], ['ir', '1112190608'], ['ir', '1337195586'], ['ir', '1148175292'], ['ir', '306106212'], ['ir', '1334767397'], ['ir', '1490924360'], ['nz', '1553335461'], ['nz', '1459959514'], ['nz', '1540849480'], ['nz', '1322200189'], ['nz', '1550688418'], ['nz', '1231214743'], ['nz', '1200361736'], ['nz', '278981407'], ['nz', '1450994021'], ['nz', '1503645220'], ['nz', '1508985962'], ['nz', '1332566161'], ['nz', '1553386057'], ['nz', '1484511465'], ['nz', '840986946'], ['nz', '213334232'], ['nz', '1553423837'], ['nz', '1479973402'], ['nz', '1552027709'], ['nz', '160904630'], ['nz', '1333552422'], ['nz', '1442950743'], ['nz', '1258471951'], ['nz', '863897795'], ['nz', '1544276107'], ['nz', '1385379989'], ['nz', '1552975779'], ['nz', '1494350511'], ['nz', '1505689225'], ['nz', '1552058389'], ['nz', '1440133626'], ['nz', '1468033698'], ['nz', '1481465172'], ['nz', '135067274'], ['nz', '1229518712'], ['nz', '174829950'], ['nz', '1545953110'], ['nz', '1264843400'], ['nz', '208011837'], ['nz', '201671138'], ['nz', '1272970334'], ['nz', '1247311946'], ['nz', '1491701052'], ['nz', '1407451189'], ['nz', '1445358795'], ['nz', '1539353861'], ['nz', '394775318'], ['nz', '1451864362'], ['nz', '73330895'], ['nz', '73802620'], ['nz', '1435516849'], ['nz', '1554782191'], ['nz', '1353058891'], ['nz', '733163012'], ['nz', '1550543719'], ['nz', '1184022695'], ['nz', '596047499'], ['nz', '1028908750'], ['nz', '1047335260'], ['nz', '1464919521'], ['nz', '1040481893'], ['nz', '1434243584'], ['nz', '1428704212'], ['nz', '343580439'], ['nz', '286123041'], ['nz', '94688506'], ['nz', '1531981810'], ['nz', '151230264'], ['nz', '1513489117'], ['nz', '1554132477'], ['nz', '1466184341'], ['nz', '1502954097'], ['nz', '1541330058'], ['nz', '342735925'], ['nz', '256580326'], ['nz', '928350893'], ['nz', '1553002053'], ['nz', '1259053911'], ['nz', '294055449'], ['nz', '1033048640'], ['nz', '175135787'], ['nz', '1507423923'], ['nz', '1495750942'], ['nz', '354668519'], ['nz', '173001861'], ['nz', '1205030005'], ['nz', '1030050704'], ['nz', '1220808096'], ['nz', '1438816586'], ['nz', '1551162705'], ['nz', '1533629268'], ['nz', '1335814741'], ['nz', '265307784'], ['nz', '1229434818'], ['nz', '978187550'], ['nz', '1533612473'], ['nz', '1500477470'], ['nz', '380627890'], ['nz', '1228993643'], ['nz', '1405950854'], ['nz', '1474245040'], ['nz', '1480311435'], ['nz', '1118188207'], ['nz', '1487513861'], ['nz', '1478211900'], ['nz', '1150510297'], ['nz', '1217681421'], ['nz', '1305527677'], ['nz', '1506906961'], ['nz', '1087147821'], ['nz', '1549903604'], ['nz', '1163907594'], ['nz', '1334767397'], ['nz', '1481398762'], ['nz', '582272991'], ['nz', '275699983'], ['nz', '809264944'], ['nz', '1478460758'], ['nz', '188674007'], ['nz', '214089682'], ['nz', '1479924698'], ['nz', '444665343'], ['nz', '1119389968'], ['nz', '587347784'], ['nz', '1050832450'], ['nz', '1552687312'], ['nz', '1492759205'], ['nz', '1436030381'], ['nz', '1508914142'], ['nz', '412151541']]

In [73]:
#############################  This code will create a file name podcast_reviews.csv #### Run this code
import csv
fields = ['id', 'country', 'review_title', 'review_text', 'review_rating']
filename = "podcast_reviews.csv"
with open(filename, 'w', newline ='',encoding="utf-8") as csvfile: 
    writer = csv.writer(csvfile) 
    writer.writerow(fields) 
###############################   

In [78]:
import json
from urllib.request import urlopen
import csv
import re 
#https://itunes.apple.com/us/rss/customerreviews/id=1222114325/json  - This URL gets only the first page which has 50 reviews

#Use this URL to find more pages of review.  By altering page=1 variable, to say page=2 you can get the 2nd page of 50 reviews
#
#findreviews = "https://itunes.apple.com/us/rss/customerreviews/page=1/id=" + qq + "/sortby=mostrecent/json"
#
 
def findReviews(pageNum , podcastList):
    
    filename = "podcast_reviews.csv"
    rows = []
    for i in podcastList: 
        country_code = i[0]
        podcast_id = i[1]
        rv = "https://itunes.apple.com/" + \
                country_code + "/rss/customerreviews/page=" + \
                str(pageNum) +"/id="+ str(podcast_id) +"/sortby=mostrecent/json"
        
        #print (rv)
        try:
            with urlopen(rv) as data_file:    
                reviewdata = json.load(data_file)
                reviewdatafeed = reviewdata["feed"] 

                try:
                    qq = reviewdatafeed["entry"]
                    for eachitem in qq:
                        rvwHeading=re.sub(r'[^A-Za-z0-9 \.]+', '', str(eachitem["title"]["label"])) 
                        rvwText= re.sub(r'[^A-Za-z0-9 \.]+', '',str(eachitem["content"]["label"]))
                        rvwRating = str(eachitem["im:rating"]["label"])
                        x = [podcast_id, country_code, rvwHeading, rvwText,rvwRating]
                        rows.append(x)
                except Exception as err:
                    pass
        except Exception as err:
            pass

        with open(filename, 'a', newline ='',encoding="utf-8") as csvfile: 
                writer = csv.writer(csvfile)
                for row in rows:
                    writer.writerow(row)

# Here I am checking for page 1 and 2 of all podcast reviews and writing to the podcast_reviews.csv file.  If we need 1000 reviews of each podcast, we need to change the range to (3,51) in the following line and run this block

In [79]:
for pgNum in range(1,3):
    findReviews(pgNum, all_podcasts[0:1])

# ENDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDD ############### Ignore all the following lines

In [26]:
stopwords = ['i', 'me', 'my', 'myself', 'we', 'our', 'ours', 'ourselves', 'you',
             "you're", "you've", "you'll", "you'd", 'your', 'yours', 'yourself', 'yourselves', 
             'he', 'him', 'his', 'himself', 'she', "she's", 'her', 'hers', 'herself', 'it', "it's",
             'its', 'itself', 'they', 'them', 'their', 'theirs', 'themselves', 'what', 'which', 'who', 'whom', 
             'this', 'that', "that'll", 'these', 'those', 'am', 'is', 'are', 'was', 'were', 'be', 'been', 'being',
             'have', 'has', 'had', 'having', 'do', 'does', 'did', 'doing', 'a', 'an', 'the', 'and', 'but', 'if', 'or',
             'because', 'as', 'until', 'while', 'of', 'at', 'by', 'for', 'with', 'about', 'against', 'between', 'into', 
             'through', 'during', 'before', 'after', 'above', 'below', 'to', 'from', 'up', 'down', 'in', 'out', 'on',
             'off', 'over', 'under', 'again', 'further', 'then', 'once', 'here', 'there', 'when', 'where', 'why',
             'how', 'all', 'any', 'both', 'each', 'few', 'more', 'most', 'other', 'some', 'such', 'no', 'nor', 'not',
             'only', 'own', 'same', 'so', 'than', 'too', 'very', 's', 't', 'can', 'will', 'just', 'don', "don't", 'should', 
             "should've", 'now', 'd', 'll', 'm', 'o', 're', 've', 'y', 'ain', 'aren', "aren't", 'couldn', "couldn't", 'didn',
             "didn't", 'doesn', "doesn't", 'hadn', "hadn't", 'hasn', "hasn't", 'haven', "haven't", 'isn', "isn't", 'ma',
             'mightn', "mightn't", 'mustn', "mustn't", 'needn', "needn't", 'shan', "shan't", 'shouldn', "shouldn't", 'wasn',
             "wasn't", 'weren', "weren't", 'won', "won't", 'wouldn', "wouldn't",
             'a', 'able', 'about', 'across', 'after', 'all', 'almost', 'also', 'am', 'among', 'an', 'and', 'any',
              'are', 'as', 'at', 'be', 'because', 'been', 'but', 'by', 'can', 'cannot', 'could', 'dear', 'did',
              'do', 'does', 'either', 'else', 'ever', 'every', 'for', 'from', 'get', 'got', 'had', 'has', 'have',
              'he', 'her', 'hers', 'him', 'his', 'how', 'however', 'i', 'if', 'in', 'into', 'is', 'it', 'its',
              'just', 'least', 'let', 'like', 'likely', 'may', 'me', 'might', 'most', 'must', 'my', 'neither',
              'no', 'nor', 'not', 'of', 'off', 'often', 'on', 'only', 'or', 'other', 'our', 'own', 'rather',
              'said', 'say', 'says', 'she', 'should', 'since', 'so', 'some', 'than', 'that', 'the', 'their',
              'them', 'then', 'there', 'these', 'they', 'this', 'tis', 'to', 'too', 'twas', 'us', 'wants',
              'was', 'we', 'were', 'what', 'when', 'where', 'which', 'while', 'who', 'whom', 'why', 'will',
              'with', 'would', 'yet', 'you', 'your', 'please', 'podcast', 'love', 'episode', 'host', 'recommend',
              'great', 'never', 'continue','review', 'reviewers','series', 'listen', 'car', 'looking', 'forward',
              'interviews', 'thanks', 'amazing', 'great', 'show', 'awesome', 'sound', 'podcasts','even', 'though',
              'story', 'episodes', 'stories', 'audiobook', 'entire', 'want'
            ]



In [15]:
def clean_text(text):
    clean_text = text.lower()
    clean_text = ''.join([c for c in clean_text if c.isalpha() or c.isspace()])
    return clean_text


In [ ]:

word_list = []

www = []
for line in word_list:
    for w in line.split():
        ww = clean_text(w)
        if ww not in stopwords:
            www.append( ww)
print (set(www))